packages/typo3-guides-extension/src/Command/RunDecorator.php

```diff
@@ -76,7 +100,149 @@ protected function execute(InputInterface $input, OutputInterface $output): int
+                $directory->getRelativePathname(),
+                $baseInputDirectives,
+                $input,
+                $output
+            );
+
+            if ($singleLocalizationExecution !== Command::SUCCESS) {
+                return $singleLocalizationExecution;
+            }
+
+        }
+
+        return Command::SUCCESS;
+    }
+
+    /**
+     * @param array<string, mixed> $baseInputDirectives
+     */
+    public function renderSingleLocalization(string $availableLocalization, array $baseInputDirectives, InputInterface $input, OutputInterface $output): int
+    {
+        $localInputDirectives = [];
+        foreach($baseInputDirectives as $baseInputDirectiveKey => $baseInputDirectiveValue) {
+            $localInputDirectives[$baseInputDirectiveKey] = $baseInputDirectiveValue . DIRECTORY_SEPARATOR . $availableLocalization;
+        }
+        $output->writeln(sprintf('<info>Trying to render %s ...</info>', $availableLocalization));
+
+        $guessInput = $this->guessInput($localInputDirectives['input-file'], $output);
+        if ($guessInput === []) {
+            $output->writeln('<info>Skipping, no entrypoint for localization found.</info>');
+            return Command::SUCCESS;
+        }
+
+        // Re-wire the command arguments to what we need for localization ...
+        $input->setArgument('input', $guessInput['input']);
+        $input->setOption('input-format', $guessInput['--input-format']);
+        $input->setOption('output', $localInputDirectives['output']);
+        $input->setOption('config', $localInputDirectives['config']);
+        $input->setOption('localization', $availableLocalization);
+
+        if ($output->isDebug()) {
+            $readableOutput = "<info>baseInputDirectives:</info>\n";
+            $readableOutput .= print_r($baseInputDirectives, true);
+            $readableOutput .= "<info>localInputDirectives:</info>\n";
+            $readableOutput .= print_r($localInputDirectives, true);
+            $readableOutput .= "<info>Entry file:</info>\n";
+            $readableOutput .= print_r($guessInput, true);
+            $readableOutput .= "<info>Actual Arguments:</info>\n";
+            $readableOutput .= print_r($input->getArguments(), true);
+            $readableOutput .= "<info>Actual Options:</info>\n";
+            $readableOutput .= print_r($input->getOptions(), true);
+            $output->writeln(sprintf("<info>DEBUG</info> Using parameters:\n%s", $readableOutput));
+        }
+
+        $processArguments = array_merge(['env', 'php', $_SERVER['PHP_SELF']], $this->retrieveLocalizationArgumentsFromCurrentArguments($input));
+
+        $process = new Process($processArguments);
+        $output->writeln(sprintf('<info>SUB-PROCESS:</info> %s', $process->getCommandLine()));
+        $result = $process->run();
+
+        $output->writeln($process->getOutput());
+        if (!$process->isSuccessful()) {
+            $output->writeln($process->getErrorOutput());
+        }
```

Comment:
Maybe you can improve this by using: 
https://symfony.com/doc/current/components/process.html#getting-real-time-process-output

Reply:
Indeed, I refactored it to directly utilize output, thanks!